pkg/trace/api/normalizer.go

```diff
@@ -134,31 +168,31 @@ func normalize(s *pb.Span) error {
 // * rejects traces where at least one span cannot be normalized
 // * return the normalized trace and an error:
 //   - nil if the trace can be accepted
-//   - an error string if the trace needs to be dropped
-func normalizeTrace(t pb.Trace) error {
+//   - a reason tag explaining the reason the traces failed normalization
+func normalizeTrace(ts *info.TagStats, t pb.Trace) error {
 	if len(t) == 0 {
-		return errors.New("empty trace")
+		atomic.AddInt64(&ts.TracesDropped.EmptyTrace, 1)
+		return errors.New("(reason:empty_trace)")
 	}
 
 	spanIDs := make(map[uint64]struct{})
-	traceID := t[0].TraceID
+	firstSpan := t[0]
 
 	for _, span := range t {
-		if span.TraceID == 0 {
-			return errors.New("empty `TraceID`")
+		if span.TraceID != firstSpan.TraceID {
+			atomic.AddInt64(&ts.TracesDropped.ForeignSpan, 1)
+			return fmt.Errorf("(reason:foreign_span): %s", span)
```

Comment:
```suggestion
			return fmt.Errorf("foreign span in trace (%d != %d): %s", span.SpanID, firstSpan.TraceID, span)
```
Let's make them all crystal clear for a human. Error logs don't need to be consistent with stats terminology.

Reply:
Same issue as https://github.com/DataDog/datadog-agent/pull/3738#discussion_r298042193, resolving to maintain one discussion.